src/js/containers/RPEInput.tsx

```diff
@@ -0,0 +1,36 @@
+import * as React from "react";
+
+export type Validator<T> = (value: T) => string | null;
+export type ChangeHandler<T> = (id: string, value: T) => void;
+
+export interface RPEInputProps<T> {
+    id: string;
+    label: string;
+    value: T;
+    validator: Validator<T>;
+    onChange: ChangeHandler<T>;
+
+    className?: string;
+
+    inputProps?: React.InputHTMLAttributes<HTMLInputElement>;
+}
+
+export function RPEInput<T extends string | number | string[]>(props: RPEInputProps<T>) {
+
+    const validationError = props.validator(props.value);
+
+    const handleChange = (ev: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>) => {
+
+        const input: HTMLInputElement | HTMLSelectElement = ev.target;
+
+        props.onChange(props.id, input.value as T);
+    };
+
+    return <div className={`input-row given-weight ${validationError ? "error" : ""} ${props.className || ""}`}>
```

Comment:
This component is called `RPEInput` but it has the class `given-weight`. The existing code uses different `given-` classes for each input. There seems to be a bug in your code that's preventing errors showing up when the user enters a too-low number for target RPE. I haven't looked too deeply, but I'm wondering if this has to do with it.

Reply:
I'll look into this, thanks